dnd5esheets/front/src/store/index.ts

```diff
@@ -1,491 +1,358 @@
+      ].map(([skill, secondary]) => ({
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `data.scores.${skill}_mod := data.scores.${secondary}_mod + data.proficiencies.skills.${skill} * data.proficiency_bonus`,
+          character
+        ),
+      })),
 
-export default function useStore() {
-  return [
-    characters,
-    {
-      update: (characterSlug: string, update: Partial<CharacterSchema>) => {
-        setCharacters(
-          characterSlug,
-          reconcile({
-            ...characters[characterSlug],
-            ...update,
-            data: {
-              ...characters[characterSlug].data,
-              ...update.data,
-              scores: {
-                ...characters[characterSlug].data.scores,
-                ...update.data?.scores,
-              },
-              hp: {
-                ...characters[characterSlug].data.hp,
-                ...update.data?.hp,
-              },
-              hit_dice: {
-                ...characters[characterSlug].data.hit_dice,
-                ...update.data?.hit_dice,
-              },
-              money: {
-                ...characters[characterSlug].data.money,
-                ...update.data?.money,
-              },
-              spells: {
-                ...characters[characterSlug].data.spells,
-                ...update.data?.spells,
-              },
-              proficiencies: {
-                ...characters[characterSlug].data.proficiencies,
-                ...update.data?.proficiencies,
-                skills: {
-                  ...characters[characterSlug].data.proficiencies.skills,
-                  ...update.data?.proficiencies?.skills,
-                },
-                saves: {
-                  ...characters[characterSlug].data.proficiencies.saves,
-                  ...update.data?.proficiencies?.saves,
-                },
-              },
-            },
-          })
-        )
+      // initiative effect
+      {
+        name: 'base',
+        priority: 10,
+        ...computeEffect(`initiative := data.scores.dexterity_mod`, character),
       },
-    },
-  ] as const
+
+      // passive perception effect
+      {
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `passive_perception := data.scores.perception_mod`,
+          character
+        ),
+      },
+
+      // TODO the effect language and CharacterSchema currently doesn't allow to write spell DC and attack bonus.
+      // // spell DC effect
+      // {
+      //   name: 'base',
+      //   priority: 10,
+      //   ...computeEffect(`spell_dc := 8 + data.scores[data.spells.spellcasting_ability] + data.proficiency_bonus`, character)
+      // },
+
+      // // spell attack bonus effect
+      // {
+      //   name: 'base',
+      //   priority: 10,
+      //   ...computeEffect(`spell_attack_bonus := data.scores[data.spells.spellcasting_ability] + data.proficiency_bonus`, character)
+      // },
```

Comment:
Doesn't https://github.com/brouberol/5esheets/pull/141/files#diff-3d43e520a4150179dbe3399da2dd4830fbb0ab5348553345569925396c22ea04R71 indicate that we now can do this?

Reply:
Will be fixed by https://github.com/brouberol/5esheets/issues/149